Source/BluetoothStateManager.swift

```diff
@@ -0,0 +1,42 @@
+import RxSwift
+
+protocol BluetoothStateProvider {
+    /// Observable which emits state changes of central manager after subscriptions
+    var rx_didUpdateState: Observable<BluetoothState> { get }
+
+    /// Current state of Central Manager
+    var state: BluetoothState { get }
+}
+
+protocol BluetoothStateManager {
+    var stateProvider: BluetoothStateProvider { get }
+}
```

Comment:
Why did you choose another level of indirection behind BluetoothStateManager? Is it really necessary?